packages/edit-site/src/components/editor/index.js

```diff
@@ -69,17 +70,30 @@ function Editor( { settings: _settings } ) {
 		setSettings,
 	] );
 
-	const { isFullscreenActive } = useSelect( ( select ) => {
+	const {
+		isFullscreenActive,
+		focusMode,
+		deviceType,
+		hasFixedToolbar,
+	} = useSelect( ( select ) => {
+		const { isFeatureActive, __experimentalGetPreviewDeviceType } = select(
+			'core/edit-site'
+		);
 		return {
-			isFullscreenActive: select( 'core/edit-site' ).isFeatureActive(
-				'fullscreenMode'
-			),
+			hasFixedToolbar: isFeatureActive( 'fixedToolbar' ),
+			isFullscreenActive: isFeatureActive( 'fullscreenMode' ),
+			focusMode: isFeatureActive( 'focusMode' ),
+			deviceType: __experimentalGetPreviewDeviceType(),
 		};
 	}, [] );
 
-	const deviceType = useSelect( ( select ) => {
-		return select( 'core/edit-site' ).__experimentalGetPreviewDeviceType();
-	}, [] );
+	useEffect( () => {
+		setSettings( {
+			...settings,
+			focusMode,
+			hasFixedToolbar,
+		} );
+	}, [ focusMode, hasFixedToolbar ] );
```

Comment:
Why is this needed?

Reply:
the block editor subscribes to settings updates via context, which is updated via the the settings state here. if we want to keep these settings in that "source of truth", then we need to do a "map props to state" sort of thing.

Alternatively, we could merge these values into the settings object in the block editor component as it renders without adding them to the settings state anywhere.

Reply:
moved it to block-editor, which makes more sense as it is a block editor setting